### ISSUE TITLE
🚀 Feature: Appwrite functions work again and do not lose data

### ISSUE THREAD

**Issue by tuvaergun** [appwrite/appwrite#6369]
### 🔖 Feature description

I am experiencing data loss. This caused serious damage.
I want to run the Executions Appwrite functions again. 
<img width="1278" alt="Screenshot 2023-09-30 at 10 45 13" src="https://github.com/appwrite/appwrite/assets/3847510/66fe8db3-5c9d-4854-8ea9-faaeb9817c16">

### 🎤 Pitch

There should be a feature that I can run again if there is an error. Or there may be a different feature such as if there is an error, try it for the 2nd or 3rd time after 1 hour.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by tripolskypetr** [comment]
Also why do you connected CI/CD from main branch to cloud? That's OK if we seeing some bugs cause It is beta but for the whole week by my app is unusable

![3cf22d9d-af7a-499a-bb74-8a742ad248c9](https://github.com/appwrite/appwrite/assets/19227776/ef5d7aea-8402-4303-8b17-b12440a17031)

**Comment by safwanyp** [comment]
@tuvaergun This will probably require a re-work of the functions, to make sure that the function's execution payload is saved after the exec fails. I don't think it stores the payload for the execution right now.
___
@tripolskypetr Do you mean connecting your GitHub repo to Appwrite? Well, the branch is selected by you when connecting to GitHub while creating a function. 

Keep in mind that while it does connect to `main`, you can make create a new branch in GitHub. When you push to this new branch, the function will build the code automatically. Since it's a branch commit, you will need to activate the build as an "active" deployment from the console.

**Comment by tripolskypetr** [comment]
@safwanyp Nope, the cloud is updating too often. Looks like spamming commits to the main branch which triggers CI/CD. For example, Realtime is down for several hours.

![image](https://github.com/appwrite/appwrite/assets/19227776/8a781359-ca59-4fd4-a307-9460961b34ca)

**Comment by tripolskypetr** [comment]
@safwanyp 
Also, [ELK stack for functions](https://www.elastic.co/elastic-stack) payload will be real good solution for @tuvaergun

**Comment by stnguyen90** [comment]
Related:

* https://github.com/appwrite/appwrite/issues/4646

**Comment by lubritto** [comment]
Any update here

**Comment by Horsty80** [comment]
I would love to have a "retry" feature ! because I need to have a workaround "save payload" mechanism to allow me retry failed execution 
sometimes failed are just flaky, and a retry can work to have a success =/
